doc/contributing/maintaining-single-executable-application-support.md

```diff
@@ -0,0 +1,110 @@
+# Maintaining Single Executable Applications support
+
+Support for [single executable applications](https://github.com/nodejs/node/blob/master/doc/contributing/technical-priorities.md#single-executable-applications)
+is one of the key technical priorities identified for the success of Node.js.
+
+## High level strategy
+
+From the [next-10 discussions](https://github.com/nodejs/next-10/blob/main/meetings/summit-nov-2021.md#single-executable-applications)
+there are 2 approaches the project believes are important to support:
+
+* Compile with Node.js into executable (`boxnode` approach).
+* Bundle into existing Node.js executable (`pkg` approach).
+
+### Compile with node into executable
+
+No additional code within the Node.js project is needed to support the
+option of compiling a bundled application along with Node.js into a single
+executable application.
+
+### Bundle into existing Node.js executable
+
+The project does not plan to provide the complete solution but instead the key
+elements which are required in the Node.js executable in order to enable
+bundling with the pre-built Node.js binaries. This includes:
+
+* Looking for a segment within the executable that holds bundled code.
+* Running the bundled code when such a segment is found.
+
+It is left up to external tools/solutions to:
+
+* Bundle code into a single script that can be executed with `-e` on
+  the command line.
+* Generate a command line with appropriate options, including `-e` to
+  run the bundled script.
+* Add a segment to an existing Node.js executable which contains
+  the command line and appropriate headers.
+* Re-generate or removing signatures on the resulting executable
+* Provide a virtual file system, and hooking it in if needed to
+  support native modules or reading file contents.
+
+## Maintaining
+
+### Compile with node into executable
+
+There are no special considerations needed.
```

Comment:
The approach of compiling with node into an executable requires that we maintain a
stable [em-bedder API](https://nodejs.org/dist/latest/docs/api/embedding.html).